kotlinlib/src/mill/kotlinlib/KotlinModule.scala

```diff
@@ -102,19 +102,48 @@ trait KotlinModule extends JavaModule { outer =>
     )().toSeq ++ kotlinWorkerClasspath()
   }
 
+  /** Flag to use the embeddable kotlin compiler */
+  def kotlinCompilerEmbeddable: Boolean = false
```

Comment:
I think this could and should be a `Task[Boolean]`, with the benefit, that it can depend on other tasks to make the decision. I could imagine so logic to analyze the plugins for compatibility, or to refer to some compatibility matrix.

Reply:
thanks! do you know why when I switch it to a task, the underlying tasks are executed as well?

e.g. in

```scala
if (kotlinCompilerEmbeddable())
   thisTaskEvalIsExecuted()
else
   thisTaskEvalIsAlsoExecuted()
```

Reply:
You're example is expanded to the following: 

```scala
Task {
   if (kotlinCompilerEmbeddable())
      thisTaskEvalIsExecuted()
   else
      thisTaskEvalIsAlsoExecuted()
   }
```

It's a task with three dependencies. Since Mill calculated the dependency graph before it starts to evaluate it, there is no way to make the graph dependent on the results of task dependencies.

But in this PR, I can't see any downsteam task where this limitation could causes any issue.

Reply:
no, I had this issue on the other PR, for KSP

Reply:
Ok, I didn't recognized that. Let me check.

Reply:
I checked. You probably meant the feature switch to enable KSP. The nice thing is, that by removing the initally proposed feature switch and implementing the `KspModule` as a stackable trait, we won't have this issue. It's now encode in the types. We change the task graph by adding the `KspModule` or not. Sounds ok to me.

Reply:
To elaborate, your initial feature switch was a on/off decision resulting in different work to do, whereas this task here is a A/B decision resulting in the same kind of work on different data.

Reply:
that was my understanding, but I'm new here and wanted to clarify :D 
either that or I was going crazy, lol

thanks for the info!